common/src/main/java/com/wynntils/services/itemfilter/SearchQuery.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.itemfilter;
+
+import com.wynntils.core.components.Services;
+import com.wynntils.models.items.WynnItem;
+import com.wynntils.utils.type.ErrorOr;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.IntStream;
+
+public class SearchQuery {
```

Comment:
So maybe you have already been discussing this, and I'm coming in late to the party. But I think this name is too generic, and would rather see it named `ItemSearchQuery`.

Reply:
I received no comments about this but it is a better name indeed.